userspace/libscap/scap-int.h

```diff
@@ -112,13 +112,32 @@ typedef struct scap_tid
 
 typedef enum ppm_reader_type
 {
-	RT_FILE = 0
+	RT_FILE = 0,
+	RT_CAPTURE_PLUGIN = 1
 } ppm_reader_type;
 
+#define READER_SEEK_BUF_SIZE    16      ///< threshold for seeking backwards for non-file readers
+#define READER_SEEK_SKIP_SIZE   2048    ///< size of the buffer used to seek forward for non-file readers
+
 struct scap_reader
 {
 	ppm_reader_type m_type;
-	gzFile m_file;
+	union {
```

Comment:
If someone else suggested using a union, ignore me, but why not just have them all part of the scap_reader struct without the union? We have plenty of other structs/classes that work this way (only some instance vars used for a given object).

Reply:
So that we can have a semblance of separation between the individual implementations? If you're worried about consistency, I'd strongly advocate separating the other structs into unions, not the other way round.

Reply:
I totally agree with @gnosek :+1: